engine/apps/slack/scenarios/manual_incident.py

```diff
@@ -0,0 +1,658 @@
+from apps.slack.slack_client.exceptions import SlackAPIException
+
+MANUAL_INCIDENT_TEAM_SELECT_ID = "manual_incident_team_select"
+MANUAL_INCIDENT_ORG_SELECT_ID = "manual_incident_org_select"
+MANUAL_INCIDENT_ROUTE_SELECT_ID = "manual_incident_route_select"
+MANUAL_INCIDENT_TITLE_INPUT_ID = "manual_incident_title_input"
+MANUAL_INCIDENT_MESSAGE_INPUT_ID = "manual_incident_message_input"
+
+DEFAULT_TEAM_VALUE = "default_team"
+
+
+# StartCreateIncidentFromMessage triggers creation of a manual incident from the slack message via submenu
+class StartCreateIncidentFromMessage(scenario_step.ScenarioStep):
+    callback_id = [
+        "incident_create",
+        "incident_create_staging",
+        "incident_create_develop",
+    ]
+
+    def process_scenario(self, slack_user_identity, slack_team_identity, payload, action=None):
+        input_id_prefix = _generate_input_id_prefix()
+
+        channel_id = payload["channel"]["id"]
+        try:
+            image_url = payload["message"]["files"][0]["permalink"]
+        except KeyError:
+            image_url = None
+        private_metadata = {
+            "channel_id": channel_id,
+            "image_url": image_url,
+            "message": {
+                "user": payload["message"].get("user"),
+                "text": payload["message"].get("text"),
+                "ts": payload["message"].get("ts"),
+            },
+            "input_id_prefix": input_id_prefix,
+            "with_title_and_message_inputs": False,
+            "submit_routing_uid": FinishCreateIncidentFromMessage.routing_uid(),
+        }
+
+        blocks = _get_manual_incident_initial_form_fields(
+            slack_team_identity, slack_user_identity, input_id_prefix, payload
+        )
+        view = _get_manual_incident_form_view(
+            FinishCreateIncidentFromMessage.routing_uid(), blocks, json.dumps(private_metadata)
+        )
+        self._slack_client.api_call(
+            "views.open",
+            trigger_id=payload["trigger_id"],
+            view=view,
+        )
+
+
+# FinishCreateIncidentFromMessage creates a manual incident from the slack message via submenu
+class FinishCreateIncidentFromMessage(scenario_step.ScenarioStep):
+    def process_scenario(self, slack_user_identity, slack_team_identity, payload, action=None):
+        Alert = apps.get_model("alerts", "Alert")
+
+        private_metadata = json.loads(payload["view"]["private_metadata"])
+
+        channel_id = private_metadata["channel_id"]
+
+        input_id_prefix = private_metadata["input_id_prefix"]
+        selected_organization = _get_selected_org_from_payload(payload, input_id_prefix)
+        selected_team = _get_selected_team_from_payload(payload, input_id_prefix)
+        selected_route = _get_selected_route_from_payload(payload, input_id_prefix)
+
+        user = slack_user_identity.get_user(selected_organization)
+        alert_receive_channel = AlertReceiveChannel.get_or_create_manual_integration(
```

Comment:
We'll create the new manual integration for each team. May be reflect that in the integration name. Just in case both teams will configure the same slack channel for their alerts